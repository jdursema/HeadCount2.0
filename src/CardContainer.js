import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';
import './style/CardContainer.css'


const CardContainer = ({schoolData, changeClass, selectedCards}) => {
     const mappedData =  schoolData.map((data)=>{
       let type
       if(selectedCards.length > 1 && (data.location === selectedCards[0].location || data.location === selectedCards[1].location)){
         type = 'card selected'
       } else{
         type = 'card'
       }
        return <Card key={data.location} 
                     location={data.location} 
                     data={data.data}
                     changeClass={changeClass}
                     type = {type} 
               />
      });

  return (
    <div className='card-container'>
      {mappedData}
    </div>
  )
}

export default CardContainer;

CardContainer.propTypes = {
  schoolData: PropTypes.array
}
