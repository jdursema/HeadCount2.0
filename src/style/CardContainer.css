.card-container{
  background-color: #F9F9F9;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
}
