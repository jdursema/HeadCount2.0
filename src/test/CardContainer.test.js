import React from 'react';
import { shallow, mount } from 'enzyme';
import CardContainer from '../CardContainer';


describe('CardContainer Test', () => {

  it('should be defined', () => {
    const mockFunc = jest.fn();
    const cardArray = [];
    const data = [{location: 'COLORADO', data: {2004: 1, 2005: .5}}]
    const cardContainer = shallow(<CardContainer schoolData={data} changeClass={mockFunc} selectedCards={cardArray} />);

    expect(cardContainer).toBeDefined();

  })

  it('should render mulitple cards', () => {
    const mockFunc = jest.fn();
    const cardArray = [];
    const data = [{location:'COLORADO', data: {2004: 1, 2005: .5}}, {location:'SPRINGFIELD', data: {2004: 1, 2005: .5}}, {location:'ASPEN', data: {2004: 1, 2005: .5}}]

    const renderCardContainer= shallow(<CardContainer schoolData={data} changeClass={mockFunc} selectedCards={cardArray} /> )
    expect(renderCardContainer.find('Card').length).toEqual(3)

  });

  it('should match snap of cardContainer', () => {
    const mockFunc = jest.fn();
    const cardArray = [];
    const data = [{location:'COLORADO', data: {2004: 1, 2005: .5}}, {location:'SPRINGFIELD', data: {2004: 1, 2005: .5}}, {location:'ASPEN', data: {2004: 1, 2005: .5}}]

    const cardContainer= shallow(<CardContainer schoolData={data} changeClass={mockFunc} selectedCards={cardArray} /> );

    expect(cardContainer).toMatchSnapshot();
  })
});
