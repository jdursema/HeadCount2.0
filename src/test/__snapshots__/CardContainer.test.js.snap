exports[`CardContainer Test should match snap of cardContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    changeClass={[Function]}
    schoolData={
        Array [
            Object {
              "data": Object {
                "2004": 1,
                "2005": 0.5,
              },
              "location": "COLORADO",
            },
            Object {
              "data": Object {
                "2004": 1,
                "2005": 0.5,
              },
              "location": "SPRINGFIELD",
            },
            Object {
              "data": Object {
                "2004": 1,
                "2005": 0.5,
              },
              "location": "ASPEN",
            },
          ]
    }
    selectedCards={Array []} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          changeClass={[Function]}
          data={
                    Object {
                              "2004": 1,
                              "2005": 0.5,
                            }
          }
          location="COLORADO"
          type="card" />,
        <Card
          changeClass={[Function]}
          data={
                    Object {
                              "2004": 1,
                              "2005": 0.5,
                            }
          }
          location="SPRINGFIELD"
          type="card" />,
        <Card
          changeClass={[Function]}
          data={
                    Object {
                              "2004": 1,
                              "2005": 0.5,
                            }
          }
          location="ASPEN"
          type="card" />,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "COLORADO",
        "nodeType": "function",
        "props": Object {
          "changeClass": [Function],
          "data": Object {
            "2004": 1,
            "2005": 0.5,
          },
          "location": "COLORADO",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "SPRINGFIELD",
        "nodeType": "function",
        "props": Object {
          "changeClass": [Function],
          "data": Object {
            "2004": 1,
            "2005": 0.5,
          },
          "location": "SPRINGFIELD",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "ASPEN",
        "nodeType": "function",
        "props": Object {
          "changeClass": [Function],
          "data": Object {
            "2004": 1,
            "2005": 0.5,
          },
          "location": "ASPEN",
          "type": "card",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            changeClass={[Function]}
            data={
                        Object {
                                    "2004": 1,
                                    "2005": 0.5,
                                  }
            }
            location="COLORADO"
            type="card" />,
          <Card
            changeClass={[Function]}
            data={
                        Object {
                                    "2004": 1,
                                    "2005": 0.5,
                                  }
            }
            location="SPRINGFIELD"
            type="card" />,
          <Card
            changeClass={[Function]}
            data={
                        Object {
                                    "2004": 1,
                                    "2005": 0.5,
                                  }
            }
            location="ASPEN"
            type="card" />,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "COLORADO",
          "nodeType": "function",
          "props": Object {
            "changeClass": [Function],
            "data": Object {
              "2004": 1,
              "2005": 0.5,
            },
            "location": "COLORADO",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "SPRINGFIELD",
          "nodeType": "function",
          "props": Object {
            "changeClass": [Function],
            "data": Object {
              "2004": 1,
              "2005": 0.5,
            },
            "location": "SPRINGFIELD",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ASPEN",
          "nodeType": "function",
          "props": Object {
            "changeClass": [Function],
            "data": Object {
              "2004": 1,
              "2005": 0.5,
            },
            "location": "ASPEN",
            "type": "card",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
